server/lib/models/matchmaking-times.ts

```diff
@@ -154,6 +154,51 @@ export async function getPastMatchmakingTimes(
   }
 }
 
+/**
+ * Get matchmaking schedule. The schedule represents future N (where N defaults to 2) matchmaking
+ * times, starting with a first S (where S defaults to enabled) status, after date X (where X
+ * defaults to current date) for a particular matchmaking type. Additionally, all subsequent
+ * matchmaking times with the same enabled status as the one before them are filtered out.
+ */
+export async function getMatchmakingSchedule(
+  matchmakingType: MatchmakingType,
+  date = new Date(),
+  firstStatus = true,
+  depth = 2,
```

Comment:
Do we ever really need more than 2 here? Feel like making this configurable adds a ton of complexity to this query

Reply:
I don't think it adds that much complexity; it only adds an additional column that gets updated in the recursion. Unfortunately we can't really eliminate the recursion since each row needs to be compared with the one before it.

Plus, I can totally see the value in potentially displaying the longer matchmaking schedule to users in some way.